### PR TITLE
Feature/ultralytics all models for Classification

### PR DESCRIPTION
**Pull request by mayankagarwals**
Successor of https://github.com/roboflow/supervision/pull/222 

Testing colab: https://colab.research.google.com/drive/1pxIqThTJwoP8Hwu95KZ3CUUio69w3_Ig?usp=sharing

### CODE REVIEW

**Comment by SkalskiP** [comment]
Hi, @mayankagarwals 👋🏻! Could you create a small Google Colab to test that change?

File: supervision/classification/core.py [hunk 1]
```diff
@@ -40,6 +42,9 @@ def __post_init__(self) -> None:
         _validate_confidence(self.confidence, n)
 
     @classmethod
+    @deprecated(
+        "This method is deprecated and removed in 0.15.0 release. Use sv.Classifications.from_ultralytics() instead."
```
Comment (SkalskiP) [inline supervision/classification/core.py:46]: Let's make that `0.16.0`. We should give people at least 2 full release cycles to migrate. That change will be released with `0.14.0`.

File: supervision/classification/core.py [hunk 2]
```diff
@@ -65,6 +70,37 @@ def from_yolov8(cls, yolov8_results) -> Classifications:
...
+            Classifications: A new Classifications object.
+
+        Example:
+            ```python
+            >>> import cv2
+            >>> from ultralytics import YOLO, FastSAM, SAM, RTDETR
+            >>> import supervision as sv
+
+            >>> image = cv2.imread(SOURCE_IMAGE_PATH)
+            >>> model = YOLO('yolov8s.pt')
```
Comment (SkalskiP) [inline supervision/classification/core.py:91]: Let's update the example to use only detection and segmentation models here.

**Review by SkalskiP — CHANGES_REQUESTED** [review]

**Comment by mayankagarwals** [comment]
Hi @SkalskiP 

Have added a testing colab in the description. 
Have updated the version in docstring for both classification and detection . Hope that's fine

File: supervision/classification/core.py (hunk 1 above)
Comment (mayankagarwals) [reply supervision/classification/core.py]: Done!

File: supervision/classification/core.py (hunk 2 above)
Comment (SkalskiP) [reply supervision/classification/core.py:91]: That is my point exactly. We are documenting `sv.Classifications.from_ultralytics` but we are showing examples of usage with detection models. We should show classification model examples.

**Comment by SkalskiP** [comment]
@mayankagarwals 🙏🏻 please also make sure to resolve `supervision/detection/core.py` conflicts.

File: supervision/classification/core.py (hunk 2 above)
Comment (mayankagarwals) [reply supervision/classification/core.py]: Done! Although I can see only yolo models for classification on ultralytics https://docs.ultralytics.com/tasks/classify/

File: supervision/classification/core.py [hunk 3]
```diff
@@ -67,6 +72,34 @@ def from_yolov8(cls, yolov8_results) -> Classifications:
...
+        Args:
+            ultralytics_results (ultralytics.yolo.engine.results.Results): The output Results instance from ultralytics model
+
+        Returns:
+            Classifications: A new Classifications object.
+
+        Example:
+            ```python
+            >>> import cv2
+            >>> from ultralytics import YOLO, FastSAM, SAM, RTDETR
```
Comment (artyaltanzaya) [inline supervision/classification/core.py:89]: Since `FastSAM, SAM, RTDETR` not used, let's remove that.

Comment (mayankagarwals) [reply supervision/classification/core.py]: Addressed :) . Good catch

File: supervision/classification/core.py [hunk 4]
```diff
@@ -40,6 +42,9 @@ def __post_init__(self) -> None:
         _validate_confidence(self.confidence, n)
 
     @classmethod
+    @deprecated(
+        "This method is deprecated and removed in 0.16.0 release. Use sv.Classifications.from_ultralytics() instead."
```
Comment (artyaltanzaya) [inline supervision/classification/core.py:46]: can we also quickly explain why `from_yolov8` is being deprecated in favor of `from_ultralytics`?

**Review by artyaltanzaya — COMMENTED** [review]
Thank you for the updates 👍. Added two comments! One is resolved, thanks for the quick action.

File: supervision/classification/core.py (hunk 4 above)
Comment (mayankagarwals) [reply supervision/classification/core.py]: Solved! Albeit these changes belong to https://github.com/roboflow/supervision/pull/278. We will just merge both of these for next release anyway

**Comment by mayankagarwals** [comment]
@onuralpszr Couple things with the pre-commit-ci 

1. Can you help me with getting the cla part resolved 
2. https://github.com/roboflow/supervision/blob/b69d264ffe5b66a5c20d8b9cd07b113b81a98e33/.pre-commit-config.yaml#L44 
I guess this hook is more restricting than supervision's existing flake8 tests. Just FYI.

**Comment by onuralpszr** [comment]
> @onuralpszr Couple things with the pre-commit-ci
> 
> 1. Can you help me with getting the cla part resolved
> 2. https://github.com/roboflow/supervision/blob/b69d264ffe5b66a5c20d8b9cd07b113b81a98e33/.pre-commit-config.yaml#L44
>    
>    I guess this hook is more restricting than supervision's existing flake8 tests. Just FYI.

For CLA I need to ask because I did not added. For flake it I only saw long error so it is normal.  :) Also you can squash those changes into commit for make it look clean :)

**Comment by onuralpszr** [comment]
@SkalskiP  this require your attention for CLA-bot: https://github.com/cla-assistant/cla-assistant#can-i-allow-bot-user-contributions

**Comment by onuralpszr** [comment]
@mayankagarwals  CLA part has been solved, thanks to @capjamesg.

**Comment by mayankagarwals** [comment]
> For flake it I only saw long line error so it is normal. :) Also you can squash those changes for make it look clean :)

Yes yes, just FYI that there is a minor behavior change. If it's intentional, no issues

> CLA part has been solved, thanks to @capjamesg.

Amazing, thanks! @artyaltanzaya All issues are addressed :)

**Review by SkalskiP — APPROVED** [review]

**Comment by SkalskiP** [comment]
@mayankagarwals I found a small problem during the final tests. You can take a look [here](https://colab.research.google.com/drive/1dJl1c-t3VkNsZaVlR6pu9r_lpyMVt0dR#scrollTo=GYnqPUTSyhGE). Committed a [fix](https://github.com/roboflow/supervision/pull/281/commits/824b2f6dc2684fd3f108af52c8111b4dd87844ad) into your branch. It looks like we are ready to merge.

**Comment by SkalskiP** [comment]
@capjamesg thanks for solving issues with bot users 🙏🏻

**Comment by mayankagarwals** [comment]
> @mayankagarwals I found a small problem during the final tests. You can take a look [here](https://colab.research.google.com/drive/1dJl1c-t3VkNsZaVlR6pu9r_lpyMVt0dR#scrollTo=GYnqPUTSyhGE). Committed a [fix](https://github.com/roboflow/supervision/pull/281/commits/824b2f6dc2684fd3f108af52c8111b4dd87844ad) into your branch. It looks like we are ready to merge.

@SkalskiP Thanks for the fix 

That's quite a crucial fix. Apologies for the lapse on my part, it's my first commit to the classification part of codebase.  Looks like I got the semantics of the Classifications feature wrong. For some reason I thought it was class of bounding boxes classified. That's my bad. 

Your fix does solve that. My only worry is `probs.data` isn't a documented field: https://docs.ultralytics.com/reference/engine/results/#ultralytics.engine.results.Probs which could mean not long term supported. But I guess it's okay for now as their documented functions don't cover all our use cases. 

Again, thanks for the support!

**Comment by SkalskiP** [comment]
> That's quite a crucial fix. Apologies for the lapse on my part, it's my first commit to the classification part of codebase. Looks like I got the semantics of the Classifications feature wrong. For some reason I thought it was class of bounding boxes classified. That's my bad.

No worries! That's why we have multiple people looking at the code and testing. ;)

**Pull request merged**